x-pack/plugins/maps/server/plugin.ts

```diff
@@ -147,27 +147,25 @@ export class MapsPlugin implements Plugin {
       return;
     }
 
-    let routesInitialized = false;
     let isEnterprisePlus = false;
+    let lastLicenseId: string;
     const emsSettings = new EMSSettings(mapsLegacyConfig, () => isEnterprisePlus);
     licensing.license$.subscribe((license: ILicense) => {
-      const basic = license.check(APP_ID, 'basic');
-
       const enterprise = license.check(APP_ID, 'enterprise');
       isEnterprisePlus = enterprise.state === 'valid';
-
-      if (basic.state === 'valid' && !routesInitialized) {
```

Comment:
checking for validity is not necessary. Other plugins do not check license-validity before registering the routes.

Reply:
Might be carryover from legacy patterns? Thanks for the clean up